src/Maestro/tests/scenarios.md

```diff
@@ -0,0 +1,131 @@
+
+with the following body:
+```json
+{
+  "repository": "https://github.com/dotnet/arcade",
+  "branch": "riarenas/test",
+  "commit": "12345",
+  "buildNumber": "12",
+  "assets": [
+    {
+      "name": "testasset",
+      "version": "1.0",
+      "nonShipping": true,
+      "locations": [
+        {
+          "location": "alocation",
+          "type": 0
+        }
+      ]
+    }
+  ],
+  "dependencies": [
+  ]
+}
+```
+
+Test that it's possible to publish a build with assets to the BAR for versions not covered by the automated scenario tests.
+
+## Dependency Flow
+
+Most of the Maestro APIs are involved in Dependency flow. This is also the area with the most scenario test coverage:
+
+The main scenarios for dependency flow are:
+
+* Query builds and assets from the BAR
+* CRUD (Create, Read, Update, Delete):
+  * Channels
+  * Default channels
+  * Subscriptions
+  * Repository Policies
+  * Goal build times
+* Applying dependency updates and opening / Managing Pull requests
+
+### Automated Testing
+
+The scenario tests available in <https://github.com/dotnet/arcade-services/tree/master/src/Maestro/tests/Scenarios> and exercise the following paths:
+
+* Build querying
+* CRUD (Creation, Reading, Updating, Deletion) of channels
+* CRUD of Default channels
+* CRUD for repository policies to a repository + branch combination
+* CRUD + triggering of subscriptions
+* Applying dependency updates / opening / merging pull requests
+* Applying batched dependency updates from multiple repositories
+* Applying dependency updates from Arcade, which include changes to the eng/common folder
+* Adding / removing a build to/from a channel after the build has been created
+
+**NOTE:** These tests only validate the latest API version. This fact is somewhat mitigated by the fact that Darc is the main mechanism to manipulate these APIs. More coverage should be added to older API versions, but they are not critical for any known scenarios.
+
+## BarViz Scenarios
+
+Barviz is used to query interesting data from the BAR, specifically, the main scenarios include visualizing:
+
+* Available channels in the BAR, along with a list of repositories that have published to a channel
+* Flow graph for a channel
+* Details of a particular build, including its dependency graph
+
+### Automated Testing
+
+We have some automated UI testing in <https://github.com/dotnet/arcade-services/tree/master/src/Maestro/maestro-angular/e2e>, but it has failed to detect the latest issues we've seen.
+
+We have also started adding unit tests for data retrieval from the Database that should be expanded as new regressions are encountered: https://github.com/dotnet/arcade-services/tree/master/src/Maestro/Maestro.Web.Tests
+
+### Manual Testing
+
+The brunt of Barviz validation is performed manually:
+
+* Go through the website making sure no errors are rendered when browsing channels / repositories and builds:
```

Comment:
Probably a quick pass at which scenarios to look at.  It's probably hard for most folks to "go through the website" grin...  Nothing extensive, but a quick pass at it is probably good.

Maybe that's what the following sections are supposed to do?  (as you can see, I'm having a little bit of a hard time following the doc - but I think that's mostly on me)

Reply:
Added some details on how the two UI views look like